### PR TITLE
New and improved docs, with a website

### PR DESCRIPTION
**Pull request by johno**

### CODE REVIEW

**Comment by ChristopherBiscardi** [comment]
Should I PR some of the docs from https://github.com/ChristopherBiscardi/gatsby-mdx here for the gatsby integration page? or wait til it's merged?

**Comment by johno** [comment]
If you wanted to PR docs into this branch that'd be great!

File: docs/advanced/writing-a-plugin.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { Message } from 'rebass'
+
+# Writing a plugin
+
+<Message>
+  This docs page is a WIP
+</Message>
```
Comment (silvenon) [inline docs/advanced/writing-a-plugin.md]: I would like to add a documentation section for integrating retext plugins, so I'm wondering what is this page meant for?

Comment (johno) [reply docs/advanced/writing-a-plugin.md]: Idea behind this page will be to show how you can customize behavior via remark/rehype plugins. With this and a custom loader page we can illustrate the ways to customize MDX for different use cases without having to pull things into core.

Would also be great to add a page on integrating retext plugins as well. PR away on any docs, that'd be lovely. A lot of work to do on the docs side still : )

**Pull request merged**